### DISCUSSION TITLE
Ideas for enums?

### DISCUSSION THREAD

**Discussion by RWOverdijk** [openapi-ts/openapi-typescript#1279] [General]
Currently I am using nestjs swagger to createn open api spec, and then this repo to create types for the client.

This works surprisingly well, except when it comes to enums. OpenAPI strips away the info needed to make a good enum and now I am unhappy with the difference between client/server when it comes to how the type is defined.

I don't have a solution other then sharing the enum, but then I can't use the types this repository generates (because it won't match). How do you solve this?

**Comment by drwpow** [comment]
Enums are a planned upcoming edition to this library! It’s been a common ask, and a couple people have expressed interest in opening a PR to make this happen. I will get to it eventually if no one else does. Either way, will tag this discussion when a PR opens up adding this and will request feedback to see if it’s a good path forward or not.
